CIScripts/Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -1,71 +1,90 @@
+function Get-LogContent {
+    param([System.Collections.Hashtable] $LogSource)
+    $ContentGetterBody = {
+        Param([Parameter(Mandatory = $true)] [string] $From)
+        $Files = Get-ChildItem -Path $From -ErrorAction SilentlyContinue
+        $Logs = @{}
+        if (-not $Files) {
+            $Logs[$From] = "<FILE NOT FOUND>"
         } else {
-            $SourceHost = "localhost"
-        }
-
-        $ContentGetterBody = {
-            Param([Parameter(Mandatory = $true)] [string] $From)
-            $Files = Get-ChildItem -Path $From -ErrorAction SilentlyContinue
-            $Logs = @{}
-            if (-not $Files) {
-                $Logs[$From] = "WARNING: FILE NOT FOUND"
-            } else {
-                $Files | ForEach-Object {
-                    $Content = Get-Content -Raw $_
-                    $Logs[$_.FullName] = $Content
+            $Files | ForEach-Object {
+                $Content = Get-Content -Raw $_
+                $Logs[$_.FullName] = if ($Content) {
+                    $Content
+                } else {
+                    "<FILE WAS EMPTY>"
                 }
             }
-            return $Logs
         }
-        $ContentGetter = New-RemoteOrLocalClosure -Func $ContentGetterBody -Session $Session -Arguments $Path
+        return $Logs
+    }
+    Invoke-CommandRemoteOrLocal -Func $ContentGetterBody -Session $LogSource.Session -Arguments $LogSource.Path
+}
 
-        $LogCleanerBody = {
-            Param([Parameter(Mandatory = $true)] [string] $What)
-            $Files = Get-ChildItem -Path $What -ErrorAction SilentlyContinue
-            $Files | ForEach-Object {
-                Remove-Item $What
-            }
+function Clear-LogContent {
+    param([System.Collections.Hashtable] $LogSource)
+    $LogCleanerBody = {
+        Param([Parameter(Mandatory = $true)] [string] $What)
+        $Files = Get-ChildItem -Path $What -ErrorAction SilentlyContinue
+        foreach ($File in $Files) {
+            Remove-Item $File
         }
-        $LogCleaner = New-RemoteOrLocalClosure -Func $LogCleanerBody -Session $Session -Arguments $Path
+    }
+    Invoke-CommandRemoteOrLocal -Func $LogCleanerBody -Session $LogSource.Session -Arguments $LogSource.Path
+}
 
-        $SingleSource = @{
-            ContentGetter = $ContentGetter
-            LogCleaner = $LogCleaner
-            SourceHost = $SourceHost
-            SourcePath = $Path
-        }
+function Merge-Logs {
+    Param([Parameter(Mandatory = $true)] [System.Collections.Hashtable[]] $LogSources,
+          [Parameter(Mandatory = $false)] [switch] $DontCleanUp)
 
-        [Diagnostics.CodeAnalysis.SuppressMessageAttribute(
-            "PSUseDeclaredVarsMoreThanAssignments", "Sources",
-            Justification="Seems like ForEach block messes up static analysis here - this var is" +
-                "already declared before the loop"
-        )]
-        $Sources += $SingleSource
-    }
+    $LogSources | ForEach-Object {
```

Comment:
Since `$LogSources` will always have at least one element, maybe we should use `foreach` here, instead of `ForEach-Object`.

Please change this in other places as well.